projects/graph/social/social.py

```diff
@@ -45,10 +52,26 @@ def populateGraph(self, numUsers, avgFriendships):
         self.users = {}
         self.friendships = {}
         # !!!! IMPLEMENT ME
+        if numUsers <= avgFriendships:
+            print("WARNING: Number of users must be greater than average friendships")
+            return
 
         # Add users
-
+        for user in range(numUsers):
+            self.addUser(user)
         # Create friendships
+        # possible_friendships = list(combinations(range(1, numUsers+1), 2))
+        # shuffle(possible_friendships)
+        # for num in range((numUsers*avgFriendships)//2):
+        #     friendship = possible_friendships[num]
+        #     self.addFriendship(friendship[0], friendship[1])
+
+        # Refactor for adding friendships stretch
+        numFriends = 0
+        while numFriends < numUsers*avgFriendships//2:
+            friends_added = self.addFriendship(
+                randint(1, numUsers), randint(1, 10))
+            numFriends += friends_added
```

Comment:
You might want to add a check to make sure that the numbers you're passing into `addFriendship()` are not the same, I'm getting some errors of `WARNING: You cannot be friends with yourself`. And if you use for loops you might be able to avoid `WARNING: Friendship already exists` errors.